cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/SourceControlException.java

```diff
@@ -17,15 +17,17 @@
 package io.cdap.cdap.sourcecontrol.operationrunner;
 
 /**
- * Exception thrown when push operation fails in operation runner.
- * Encapsulates all underlying exceptions.
+ * Exception thrown when source control operation fails in runner.
+ * Encapsulates some underlying exceptions.
+ * Should be subclassed for common root-causes
  */
-public class PushFailureException extends Exception {
-  public PushFailureException(String message, Exception cause) {
-    super(message, cause);
+// TODO https://cdap.atlassian.net/browse/CDAP-20410 Classify and make suberrors based on root causes
+public class SourceControlException extends RuntimeException {
```

Comment:
Any reason we want it to extend `RuntimeException`?

Reply:
As we do not have any cases where we want to handle SourceControlException and return this as 500 to the user keeping this as an unchecked exception.